DeepMRSeg/deepmrseg_test.py

```diff
@@ -432,140 +461,387 @@ def _main():
+			_sys.stdout.flush()
+	#ENDFOR
+	
+	#####################################################################
+	## Merge output from all models
 	
-	#resource package only available in Unix
-	if _platform.system() != 'Windows':
-		import resource as _resource
 	
-		rus = _resource.getrusage(0)
-		print("\tutime \t:", _np.round( rus.ru_utime, 2 ))
-		print("\tstime \t:", _np.round( rus.ru_stime, 2 ))
-		print("\tmaxrss \t:", _np.round( rus.ru_maxrss / 1.e6, 2 ), "GB")
+	if num_models > 1:
 
+		print("\n")
+		print("\n---->	Creating final predictions by combining output from each model")
 		_sys.stdout.flush()
+		
+		#WITH OPENFILE
+		with open(FLAGS.sList) as f:
+			reader = _csv.DictReader( f )
+
+			#FOR
+			for row in reader:
+			
+				### Get image filenames
+				subId = row[idCol]
+				outImg = row[outCol]
+
+				if not _os.path.isfile( outImg ):
+
+					# Get image size from the first nii.gz image 
+					im_shape = []
+					for f in _os.listdir(_os.path.join(tmpDir, subId)):
+						if f.endswith('.nii.gz'):
+							break
+					im_shape = _nib.load(_os.path.join(tmpDir, subId, f)).get_data().shape
+					
+					# Create array to store output probabilities
+					val_prob = _np.zeros( ( list(im_shape[0:3]) + [trainflag_num_classes] ) )
+				
+					# Create output directory if it doesn't exist already
+					##IF
+					if not _os.path.isdir( _os.path.dirname(outImg) ):
+						_os.makedirs( _os.path.dirname(outImg) )
+					#ENDIF
+
+					#FOR
+					for clInd in range(trainflag_num_classes):
+
+						clSuff = '_probabilities_' + str(clInd)
+
+						# Combine prob values for different models
+						files = []
+						for f in _os.listdir(_os.path.join(tmpDir, subId)):
+							if f.endswith(clSuff + '.nii.gz'):
+								files.append(_os.path.join(tmpDir, subId, f))
+						
+						if len(files)>0:
+							niiTmp = _nib.load(files[0])
+							probOut = niiTmp.get_data()                
+							if len(files) > 1:
+								for tmpF in files[1:]:
+									niiTmp = _nib.load(tmpF)
+									probTmp = niiTmp.get_data()
+									probOut += probTmp
+								probOut = probOut / len(files)
+								
+							# Write probabilities
+							if FLAGS.probs:
+								outNii = _nib.Nifti1Image( probOut, niiTmp.affine, niiTmp.header )
+								outNii.to_filename(outImg.replace('.nii.gz', '_probabilities_' + str(clInd) + '.nii.gz'))
+						val_prob[:,:,:,clInd] = probOut
+					#ENDFOR
+
+					### Get preds from prob matrix
+					val_bin = _np.argmax( val_prob, axis=-1 )
+
+					# Write binary mask
+					outNii = _nib.Nifti1Image( val_bin, niiTmp.affine, niiTmp.header )
+					outNii.set_data_dtype( 'uint8' )
+					outNii.to_filename(outImg)
+
+					print("\n")
+					print("\n---->	Out image : " + outImg)
+					_sys.stdout.flush()
+				
+				#ENDIF
+			#ENDFOR
+		#ENDWITH
+	#ENDIF
+	
+	### Remove tmpDir and its contents, if not user defined
+	#IF
+	if not user_defined_tmpdir:
+		#TRY
+		try:
+			_shutil.rmtree( tmpDir )
```

Comment:
This line gives an error:
Failed to delete: /tmp/deepmrseg_test_zqkha7pf

There is a csv file in the tempdir that needs to be deleted first before running shutil.rmtree

Reply:
Fixed; shutil was not imported